test/operations/operations.go

```diff
@@ -8,74 +8,117 @@ import (
-		},
-		Spec: configSpec,
-	}
+func CreateConfig(gvk schema.GroupVersionKind, configSpec cnao.NetworkAddonsConfigSpec) {
+	By(fmt.Sprintf("Applying NetworkAddonsConfig spec:\n%s", configSpecToYaml(configSpec)))
 
+	config := &unstructured.Unstructured{}
+	config.SetGroupVersionKind(gvk)
+	config.SetName(names.OPERATOR_CONFIG)
+	config.Object["spec"] = configSpec
 	err := framework.Global.Client.Create(context.TODO(), config, &framework.CleanupOptions{})
 	Expect(err).NotTo(HaveOccurred(), "Failed to create the Config")
 }
 
-func UpdateConfig(configSpec cnao.NetworkAddonsConfigSpec) {
+func UpdateConfig(gvk schema.GroupVersionKind, configSpec cnao.NetworkAddonsConfigSpec) {
 	By(fmt.Sprintf("Updating NetworkAddonsConfig:\n%s", configSpecToYaml(configSpec)))
 
 	// Get current Config
-	config := GetConfig()
+	config := GetConfig(gvk)
 
 	// Update the Config with the desired Spec
-	config.Spec = configSpec
+	config.Object["spec"] = configSpec
 	err := framework.Global.Client.Update(context.TODO(), config)
 	Expect(err).NotTo(HaveOccurred(), "Failed to update the Config")
 }
 
-func DeleteConfig() {
+func DeleteConfig(gvk schema.GroupVersionKind) {
 	By("Removing NetworkAddonsConfig")
 
-	config := &cnaov1.NetworkAddonsConfig{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: names.OPERATOR_CONFIG,
-		},
-	}
+	config := &unstructured.Unstructured{}
+	config.SetGroupVersionKind(gvk)
+	config.SetName(names.OPERATOR_CONFIG)
 
 	err := framework.Global.Client.Delete(context.TODO(), config)
 	ExpectWithOffset(1, err).NotTo(HaveOccurred(), "Failed to remove the Config")
 
 	// Wait until the config is deleted
 	EventuallyWithOffset(1, func() error {
-		return framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: names.OPERATOR_CONFIG}, &cnaov1.NetworkAddonsConfig{})
-	}, 60*time.Second, 1*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(apierrors.IsNotFound, BeTrue())), fmt.Sprintf("should successfuly delete config '%s'", config.Name))
+		return framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: names.OPERATOR_CONFIG}, config)
+	}, 60*time.Second, 1*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(apierrors.IsNotFound, BeTrue())), fmt.Sprintf("should successfuly delete config '%s'", config.GetName()))
+
+}
+
+func GetConfigStatus(gvk schema.GroupVersionKind) *cnao.NetworkAddonsConfigStatus {
+	By("Getting the current config")
+	config := &unstructured.Unstructured{}
+	config.SetGroupVersionKind(gvk)
+	err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: names.OPERATOR_CONFIG}, config)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+
+	cnaoV1 := cnaov1.NetworkAddonsConfig{}
+	cnaoV1alpha1 := cnaov1.NetworkAddonsConfig{}
+	switch gvk {
+	case cnaoV1.GroupVersionKind():
+		return &convertToConfigV1(config).Status
+	case cnaoV1alpha1.GroupVersionKind():
+		return &convertToConfigV1alpha1(config).Status
+	}
+
+	Fail(fmt.Sprintf("gvk %v not supported", gvk))
+	return nil
```

Comment:
We cannot retrieve status from unstructured ?

Reply:
`framework.Global.Client.Status().` doesn't support `Get`,  only `Update\Patch`.
I did not find a way to get the status object out of the struct